type-definitions/ts-tests/map.ts

```diff
@@ -61,6 +71,31 @@ import { Map, List } from 'immutable';
 
   // $ExpectError
   Map<number, number>().get<number>(4, 'a');
+
+  // $ExpectType number
+  Map({ a: 4, b: true }).get('a');
+
+  // $ExpectType boolean
+  Map({ a: 4, b: true }).get('b');
+
+  // $ExpectType boolean
+  Map({ a: Map({ b: true }) })
+    .get('a')
+    .get('b');
+
+  // $ExpectError
+  Map({ a: 4 }).get('b');
+}
+
+{
+  // Minimum TypeScript Version: 4.1
+  // #getIn
+
+  // $ExpectType number
+  Map({ a: 4, b: true }).getIn(['a']);
+
+  // $ExpectType number
+  Map({ a: Map({ b: Map({ c: Map({ d: 4 }) }) }) }).getIn(['a', 'b', 'c', 'd']);
 }
```

Comment:
Can you add getIn tests where nested Map are constructed by other means (eg not only `ObjectFromMap` type?) as well as having `List` within?

Reply:
Unfortunatly, it's only working for `MapFromObject` for now.